### PR TITLE
doc(Fees): Add doc for fee advance in invoice

### PR DESCRIPTION
**Pull request by michaelponrajah**
## Description

Add documentation for regroup fee paid in advance in invoices

### CODE REVIEW

File: guide/invoicing/fees.mdx [hunk 1]
```diff
@@ -72,3 +72,23 @@ This is illustrated below.:
...
+</Info>
+
+This option allows you to define the payment status of a fee yourself. The available options are: `pending`, `succeeded`, `failed`, and `refunded`.
+
+Note that `payment_status` is used to group fees with a `succeeded` status into a specific invoice if `regroup_paid_fee` is set to `invoice`. Outside of this process, the `payment_status` is for your internal use.
+
+## Delete a fee
+
+<Info>
+  The `payment_status` of a fee can only be configured if `pay_in_advance` is
```
Comment (brunomiguelpinto) [inline guide/invoicing/fees.mdx:90]: need to change this to the delete description

File: api-reference/plans/object.mdx [hunk 2]
```diff
@@ -349,15 +354,22 @@ description: "This object represents a plan. This plan can then be assigned to a
...
+          reception. If false the fee is created at event reception but its
+          invoicing will follow the group_paid_fees settings.
         </ResponseField>
         <ResponseField name="invoiceable" type="boolean">
           This field specifies whether the charge should be included in a proper
           invoice. If set to `false`, no invoice will be issued for this charge.
           You can only set it to `false` when `pay_in_advance` is `true`.
         </ResponseField>
+        <ResponseField name="regroup_paid_fees" type="string">
+          This setting can only be configured if pay_in_advance is false. This
```
Comment (julienbourdeau) [inline api-reference/plans/object.mdx:367]:
```suggestion
          This setting can only be configured if pay_in_advance is true and invoiceable is false. This
```

File: api-reference/plans/object.mdx [hunk 3]
```diff
@@ -349,15 +354,22 @@ description: "This object represents a plan. This plan can then be assigned to a
         </ResponseField>
         <ResponseField name="pay_in_advance" type="boolean">
           This field determines the billing timing for this specific usage-based
-          charge. When set to `true`, the charge is due and invoiced
-          immediately. Conversely, when set to `false`, the charge is due and
-          invoiced at the end of each billing period.
+          charge. When set to true, the fee is created and invoices at the event
+          reception. If false the fee is created at event reception but its
+          invoicing will follow the group_paid_fees settings.
```
Comment (julienbourdeau) [inline api-reference/plans/object.mdx:359]: This is for `invoiceable` attribute.

```suggestion
          reception. If false the fee will be part of the next subscription invoice
```

File: guide/invoicing/fees.mdx [hunk 4]
```diff
@@ -72,3 +72,23 @@ This is illustrated below.:
   }
 }
 ```
+
+## Update fee payment status
+
+<Info>
+  The `payment_status` of a fee can only be configured if `pay_in_advance` is
+  `true` and `lago_invoice_id` is `null`.
```
Comment (julienbourdeau) [inline guide/invoicing/fees.mdx:80]: Technically, we don't enforce anything to `pay_in_advance`. It only check if the fee is attached to an invoice, `lago_invoice_id` must be null to update the fee.

File: guide/invoicing/fees.mdx [hunk 5]
```diff
@@ -72,3 +72,23 @@ This is illustrated below.:
...
+
+This option allows you to define the payment status of a fee yourself. The available options are: `pending`, `succeeded`, `failed`, and `refunded`.
+
+Note that `payment_status` is used to group fees with a `succeeded` status into a specific invoice if `regroup_paid_fee` is set to `invoice`. Outside of this process, the `payment_status` is for your internal use.
+
+## Delete a fee
+
+<Info>
+  The deletion of a fee can only be doable if `pay_in_advance` is `true` and
+  `lago_invoice_id` is `null`.
```
Comment (julienbourdeau) [inline guide/invoicing/fees.mdx:91]: same here, we don't check `pay_in_advance`

File: guide/plans/charges/invoiceable-vs-noninvoiceable.mdx [hunk 6]
```diff
@@ -1,33 +1,42 @@
...
+The `"invoiceable": false` option is **particularly relevant for fintech companies that want to charge customers instantly
+without issuing an invoice**. This feature can be useful for banking transactions like ATM withdrawals, FX transfers and other
 similar scenarios.
 
-By setting the invoiceable parameter to false, you can streamline the payment process, eliminating the step of generating 
-invoices for these specific charges.
+By setting the invoiceable parameter to false, a fee is created by default, and a webhook `fee.created` is sent immediately upon any changes in usage without generating an invoice. However, you can decide the charge invoicing strategy:
+
+1. `null`: Fees generated never appear on invoices (default behaviour).
+2. `in_arrears`: Groups all paid fees into one invoice at the end of the billing period.
```
Comment (julienbourdeau) [inline guide/plans/charges/invoiceable-vs-noninvoiceable.mdx:22]:
```suggestion
2. `invoice`: Groups all paid fees into one invoice at the end of the billing period.
```

File: guide/plans/charges/invoiceable-vs-noninvoiceable.mdx [hunk 7]
```diff
@@ -1,33 +1,42 @@
...
+
+Please note that if `in_arrears`, all fees generated with a `payment_status` different from `succeeded` won’t appear on invoice generated at the end of the billing period. They will appear on next ones if you manually edit their `payment_status` to `succeeded`.
```
Comment (julienbourdeau) [inline guide/plans/charges/invoiceable-vs-noninvoiceable.mdx:24]:
```suggestion
Please note that if `invoice`, all fees generated with a `payment_status` different from `succeeded` won’t appear on invoice generated at the end of the billing period. They will appear on next ones if you manually edit their `payment_status` to `succeeded`.
```

File: guide/plans/charges/invoiceable-vs-noninvoiceable.mdx [hunk 8]
```diff
@@ -1,33 +1,42 @@
...
 
-The `"invoiceable": false` option is **particularly relevant for fintech companies that want to charge customers instantly 
-without issuing an invoice**. This feature can be useful for banking transactions like ATM withdrawals, FX transfers and other 
+The `"invoiceable": false` option is **particularly relevant for fintech companies that want to charge customers instantly
+without issuing an invoice**. This feature can be useful for banking transactions like ATM withdrawals, FX transfers and other
 similar scenarios.
 
-By setting the invoiceable parameter to false, you can streamline the payment process, eliminating the step of generating 
-invoices for these specific charges.
+By setting the invoiceable parameter to false, a fee is created by default, and a webhook `fee.created` is sent immediately upon any changes in usage without generating an invoice. However, you can decide the charge invoicing strategy:
```
Comment (julienbourdeau) [inline guide/plans/charges/invoiceable-vs-noninvoiceable.mdx:19]:
```suggestion
By setting the invoiceable parameter to false, a fee is created by default, and a webhook `fee.created` is sent immediately upon any changes in usage without generating an invoice. However, you can configure the charge invoicing strategy via `regroup_paid_fees`:
```

**Pull request merged**